cmd/podman/system_df.go

```diff
@@ -0,0 +1,363 @@
+		RunE: func(cmd *cobra.Command, args []string) error {
+			dfSystemCommand.GlobalFlags = MainGlobalOpts
+			return dfSystemCmd(&dfSystemCommand)
+		},
+	}
+)
+
+func init() {
+	dfSystemCommand.Command = _dfSystemCommand
+	dfSystemCommand.SetUsageTemplate(UsageTemplate())
+}
+
+var systemDfHeader = map[string]string{
+	"Type":        "TYPE",
+	"TotalCount":  "TOTAL",
+	"Active":      "ACTIVE",
+	"Size":        "SIZE",
+	"Reclaimable": "RECLAIMABLE",
+}
+
+type systemDfDiskUsage struct {
+	Type        string
+	TotalCount  int
+	Active      int
+	Size        string
+	Reclaimable string
+}
+
+func dfSystemCmd(c *cliconfig.SystemDfValues) error {
+	runtime, err := libpodruntime.GetRuntime(&c.PodmanCommand)
+	if err != nil {
+		return errors.Wrapf(err, "Could not get runtime")
+	}
+	defer runtime.Shutdown(false)
+
+	ctx := getContext()
+
+	systemDfDiskUsages, err := getDiskUsage(ctx, runtime)
+	if err != nil {
+		return errors.Wrapf(err, "error getting output of system df")
+	}
+	out := formats.StdoutTemplateArray{Output: systemDfDiskUsageToGeneric(systemDfDiskUsages), Template: "table {{.Type}}\t{{.TotalCount}}\t{{.Active}}\t{{.Size}}\t{{.Reclaimable}}", Fields: systemDfHeader}
+	formats.Writer(out).Out()
+	return nil
+}
+
+func systemDfDiskUsageToGeneric(diskUsages []systemDfDiskUsage) (out []interface{}) {
+	for _, usage := range diskUsages {
+		out = append(out, interface{}(usage))
+	}
+	return out
+}
+
+func getDiskUsage(ctx context.Context, runtime *libpod.Runtime) ([]systemDfDiskUsage, error) {
+	images, err := runtime.ImageRuntime().GetImages()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get images")
+	}
+	containers, err := runtime.GetAllContainers()
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting all containers")
+	}
+	volumes, err := runtime.GetAllVolumes()
+	if err != nil {
+		return nil, errors.Wrap(err, "error getting all volumes")
+	}
+	activeContainers, err := activeContainers(runtime)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting active containers")
+	}
+	imagesUsedbyCtrMap, imagesUsedbyActiveCtr, err := imagesUsedbyCtr(containers, activeContainers)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting getting images used by containers")
+	}
+	volumeUsedByContainerMap := volumeUsedByContainer(containers)
+
+	imageDiskUsage, err := getImageDiskUsage(ctx, images, imagesUsedbyCtrMap, imagesUsedbyActiveCtr)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting disk usage of images")
+	}
+	containerDiskUsage, err := getContainerDiskUsage(containers, activeContainers)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting disk usage of containers")
+	}
+	volumeDiskUsage, err := getVolumeDiskUsage(volumes, volumeUsedByContainerMap)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting disk usage of volumess")
+	}
+	var systemDfDiskUsages []systemDfDiskUsage
+	systemDfDiskUsages = append(systemDfDiskUsages, imageDiskUsage)
```

Comment:
Why are you appending all of these together?

Reply:
append same struct type (same header )
`TYPE TOTAL ACTIVE SIZE RECLAIMABLE`, appned together for output 

can change it to `systemDfDiskUsages := []systemDfDiskUsage{imageDiskUsage, containerDiskUsage, volumeDiskUsage}`